src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs

```diff
@@ -369,6 +369,20 @@ public T WithCorrelationId(Guid correlationId)
             return (T)this;
         }
 
+        /// <summary>
+        /// Determines if the client app cannot access the network due to power saving optimizations or if the device is currently in idle mode
+        /// and will throw an <see cref="MsalClientException"/> with broker_client_power_opt_error as an error code.
+        /// </summary>
+        /// <param name="checkPowerOp">Check for power optimization.</param>
+        /// <returns>The builder to chain the .With methods.</returns>
+        /// <remarks> Idle mode happens when a device has been sitting unused and unmoving for a sufficiently long period of time,
+        /// so that it decides to go into a lower power-use state. This may involve things like turning off network access to apps.</remarks>
+        public T WithPowerOptimizationCheck(bool checkPowerOp)
```

Comment:
- I am leaving it as false for now but I may change it after speaking with the android team about this today.

- This check has the potential to cause issues in some scenarios according to this MSAL.Android bug here (https://github.com/AzureAD/microsoft-authentication-library-for-android/issues/998) MSAL.Android has a boolean for power opt checks so I am enabling MSAL.NET to do the same. 

- This will use the same guidance provided by MSAL android. I am having a hard time finding it in their documentation at the moment so when i sync with them today I can add I here.